crates/oxc_sourcemap/src/encode.rs

```diff
@@ -0,0 +1,138 @@
+/// Port from https://github.com/getsentry/rust-sourcemap/blob/master/src/encoder.rs
+/// It is a helper for encode `SourceMap` to vlq sourcemap string, but here some different.
+/// - Quote `source_content` at parallel.
+/// - If you using `ConcatSourceMapBuilder`, serialize `tokens` to vlq `mappings` at parallel.
+use crate::{token::TokenChunk, SourceMap, Token};
+use rayon::prelude::*;
+
+pub fn encode(sourcemap: &SourceMap) -> String {
+    let mut buf = String::new();
+    buf.push_str("{\"version\":3,");
+    if let Some(file) = sourcemap.get_file() {
+        buf.push_str("\"file\":\"");
+        buf.push_str(file);
+        buf.push_str("\",");
+    }
+    buf.push_str("\"names\":[");
+    buf.push_str(&sourcemap.names.iter().map(|x| format!("{x:?}")).collect::<Vec<_>>().join(","));
+    buf.push_str("],\"sources\":[");
+    buf.push_str(&sourcemap.sources.iter().map(|x| format!("{x:?}")).collect::<Vec<_>>().join(","));
+    // Quote `source_content` at parallel.
+    if let Some(source_contents) = &sourcemap.source_contents {
+        buf.push_str("],\"sourcesContent\":[");
+        buf.push_str(
+            &source_contents.par_iter().map(|x| format!("{x:?}")).collect::<Vec<_>>().join(","),
```

Comment:
This `par_iter` is probably redundant. Remember, it's costly to create threads.

Reply:
It is expected for larger string, quote `source content` is very expensive.

Reply:
<img width="1089" alt="image" src="https://github.com/oxc-project/oxc/assets/14008915/b4ffeccc-427d-49b7-85d6-ef8817967c32">
I upload a profile. In a threejs10x build, its time percent is `23%`.

Reply:
ref: esbuild sourcemap also do it. https://github.com/evanw/esbuild/blob/main/internal/bundler/bundler.go#L2997-L3045